File: src/main/java/biblioteca/Material.java
package biblioteca;

public abstract class Material {
    protected String titulo;
    protected String autor;
    protected EstadoMaterial estado = EstadoMaterial.DISPONIBLE;


    // Contrato declarado:
    // Pre: dias >= 1 && dias <= 30 && estado == DISPONIBLE
    // Post: estado == PRESTADO && retorna confirmación no nula


    public abstract String prestar(String dniLector, int dias);


    // Pre: estado == PRESTADO
    // Post: estado == DISPONIBLE && lectorDni == null


    public abstract void devolver();


    public EstadoMaterial getEstado() { return estado; }
}

File: biblioteca/material.py
# biblioteca/material.py

from biblioteca.libro import EstadoMaterial


class Material:
    def __init__(self, titulo: str, autor: str, anio: int):
        if anio <= 1900:
            raise ValueError('El año debe ser posterior a 1900')
        self._titulo = titulo
        self._autor = autor
        self._anio = anio
        self._estado = EstadoMaterial.DISPONIBLE

    def prestar(self, dni_lector: str, dias: int) -> str:
        raise NotImplementedError

    def devolver(self) -> None:
        raise NotImplementedError

    def get_estado(self) -> EstadoMaterial:
        return self._estado